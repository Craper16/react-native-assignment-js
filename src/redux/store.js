import {configureStore, getDefaultMiddleware} from '@reduxjs/toolkit';
import authSlice from './auth/authSlice';
import {authApi} from './api/authApi';
import articlesSlice from './articles/articlesSlice';
import {articlesApi} from './api/articlesApi';

export const store = configureStore({
  reducer: {
    auth: authSlice,
    articles: articlesSlice,
    [authApi.reducerPath]: authApi.reducer,
    [articlesApi.reducerPath]: articlesApi.reducer,
  },
  middleware: getDefaultMiddleware =>
    getDefaultMiddleware().concat(authApi.middleware).concat(articlesApi.middleware),
});

export default store;
