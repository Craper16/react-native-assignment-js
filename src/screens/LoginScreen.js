import React from 'react';
import {
  View,
  StyleSheet,
  KeyboardAvoidingView,
  ScrollView,
  Keyboard,
} from 'react-native';
import {storage} from '../storage/storage';
import {Button, Text, TextInput} from 'react-native-paper';
import {useDispatch} from 'react-redux';
import {useSignInUserMutation} from '../redux/api/authApi';
import {setUser} from '../redux/auth/authSlice';
import * as yup from 'yup';
import {Formik} from 'formik';
import {TouchableWithoutFeedback} from 'react-native-gesture-handler';

const loginValidationSchema = yup.object().shape({
  username: yup.string().required('Username must not be empty'),
  password: yup.string().required('Password must not be empty.'),
});

const LoginScreen = () => {
  const dispatch = useDispatch();
  const [signInUser, {data, isLoading, error, isError, isSuccess}] =
    useSignInUserMutation();

  const LoginUser = async () => {
    await storage.set('accessToken', data?.accessToken);
    dispatch(setUser({accessToken: data?.accessToken}));
  };

  if (isSuccess) {
    LoginUser();
  }

  return (
    <Formik
      initialValues={{username: '', password: ''}}
      validateOnMount={true}
      onSubmit={values => signInUser({...values})}
      validationSchema={loginValidationSchema}>
      {({
        handleChange,
        handleBlur,
        handleSubmit,
        values,
        touched,
        isValid,
        errors,
      }) => (
        <KeyboardAvoidingView
          behavior="padding"
          keyboardVerticalOffset={50}
          style={styles.screen}>
          <TouchableWithoutFeedback
            onPress={Keyboard.dismiss}
            style={styles.mainContainer}>
            <Text style={styles.title}>Login to view Articles!</Text>
            <View style={styles.inputContainer}>
              <ScrollView>
                <View style={styles.textInputContainer}>
                  <TextInput
                    autoCapitalize={false}
                    textColor="#8A2BE2"
                    onChangeText={handleChange('username')}
                    onBlur={handleBlur('username')}
                    value={values.username}
                    mode="outlined"
                    label="Username"
                    error={errors.username && touched.username}
                  />
                  {errors.username && touched.username && (
                    <Text style={styles.errors}>{errors.username}</Text>
                  )}
                </View>
                <View style={styles.textInputContainer}>
                  <TextInput
                    autoCapitalize={false}
                    textColor="#8A2BE2"
                    onChangeText={handleChange('password')}
                    onBlur={handleBlur('password')}
                    value={values.password}
                    mode="outlined"
                    secureTextEntry
                    label="Password"
                    error={errors.password && touched.password}
                  />
                  {errors.password && touched.password && (
                    <Text style={styles.errors}>{errors.password}</Text>
                  )}
                </View>
                <View style={styles.actions}>
                  <Button
                    disabled={!isValid || isLoading}
                    onPress={handleSubmit}
                    mode="elevated"
                    loading={isLoading}>
                    Login
                  </Button>
                </View>
                {isError && (
                  <View style={styles.apiErrorsContainer}>
                    <Text style={styles.apiErrors}>
                      {error.data?.message || error.error}
                    </Text>
                  </View>
                )}
              </ScrollView>
            </View>
          </TouchableWithoutFeedback>
        </KeyboardAvoidingView>
      )}
    </Formik>
  );
};

const styles = StyleSheet.create({
  screen: {
    flex: 1,
    backgroundColor: '#aab1eebb',
  },
  mainContainer: {
    alignItems: 'center',
    justifyContent: 'center',
    marginVertical: 100,
  },
  title: {
    fontSize: 24,
    fontWeight: 'bold',
    marginBottom: 16,
    color: '#8A2BE2',
  },
  inputContainer: {
    width: '80%',
    padding: 2,
    marginBottom: 9,
    marginVertical: 14,
  },
  textInputContainer: {
    margin: 5,
    padding: 4,
    justifyContent: 'center',
  },
  errors: {
    fontSize: 14,
    fontWeight: 'bold',
    marginTop: 5,
    color: 'red',
  },
  apiErrorsContainer: {
    width: '100%',
    justifyContent: 'center',
    alignItems: 'center',
  },
  apiErrors: {
    fontSize: 14,
    fontWeight: 'bold',
    marginTop: 7,
    color: 'red',
  },
  actions: {
    width: '100%',
    height: 48,
    justifyContent: 'center',
    alignItems: 'center',
  },
});

export default LoginScreen;
