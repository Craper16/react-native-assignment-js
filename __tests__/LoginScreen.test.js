import React from 'react';
import renderer, {act} from 'react-test-renderer';
import LoginScreen from '../src/screens/LoginScreen';
import store from '../src/redux/store';
import {fireEvent, render, waitFor} from '@testing-library/react-native';
import {Provider} from 'react-redux';
import {Provider as PaperProvider} from 'react-native-paper';

// Testing login screen main functionalities:
// if login screen renders login form, handles API response on submit
// using mock requests
// and if button is disabled when inputs are empty

describe('LoginScreen', () => {
  test('Screen renders correctly', () => {
    const tree = renderer.create(
      <Provider store={store}>
        <PaperProvider>
          <LoginScreen />
        </PaperProvider>
      </Provider>,
    );
    expect(tree).toMatchSnapshot();
  });

  test('Login screen should contain username and password input fields and login button', () => {
    const {getByTestId} = render(
      <Provider store={store}>
        <PaperProvider>
          <LoginScreen />
        </PaperProvider>
      </Provider>,
    );
    const username = getByTestId('username');
    const password = getByTestId('password');
    const loginButton = getByTestId('loginButton');

    expect(username).toBeTruthy();
    expect(password).toBeTruthy();
    expect(loginButton).toBeTruthy();
  });

  // EXECUTE LOGIN API FAIL AND SUCCESS

  test('API is called after pressing login button and it handles incorrect credentials', async () => {
    const {getByTestId, getByText} = render(
      <Provider store={store}>
        <PaperProvider>
          <LoginScreen />
        </PaperProvider>
      </Provider>,
    );
    const username = getByTestId('username');
    const password = getByTestId('password');
    const loginButton = getByTestId('loginButton');

    await act(async () => {
      await fireEvent.changeText(username, 'wrongusername');
      await fireEvent.changeText(password, 'wrongpassword');
      await fireEvent.press(loginButton, {username, password});
    });

    // if you put correct credentials test will fail

    await waitFor(() => {
      expect(getByText('Please check your login credentials')).toBeTruthy();
      expect(store.getState().auth.accessToken).not.toBe('mocktoken');
    });
  });

  test('API is called after pressing login button and is successful', async () => {
    const {getByTestId} = render(
      <Provider store={store}>
        <PaperProvider>
          <LoginScreen />
        </PaperProvider>
      </Provider>,
    );
    const username = getByTestId('username');
    const password = getByTestId('password');
    const loginButton = getByTestId('loginButton');

    await act(async () => {
      await fireEvent.changeText(username, 'candidate');
      await fireEvent.changeText(password, 'P@ssw0rd');
      await fireEvent.press(loginButton, {username, password});
    });

    // In my code accessToken is stored in redux and if it exists
    // and it's not undefined or null
    // It redirects/changes the react navigation stack to the articles/dashboard screen
    // So here im testing if the accessToken is the same as my msw response token
    // to test redirect
    // whys this comment too long

    await waitFor(() => {
      expect(store.getState().auth.accessToken).toBe('mocktoken');
    });
  });

  // EXECUTE LOGIN API FAIL AND SUCCESS ^

  test('Button is disabled when both input fields are empty', async () => {
    const {getByTestId, getByText} = render(
      <Provider store={store}>
        <PaperProvider>
          <LoginScreen />
        </PaperProvider>
      </Provider>,
    );

    const username = getByTestId('username');
    const password = getByTestId('password');
    const loginButton = getByTestId('loginButton');

    await act(async () => {
      await fireEvent.changeText(username, 'test');
      await fireEvent.changeText(password, 'test');
    });

    expect(loginButton.props.accessibilityState).toStrictEqual({
      disabled: false,
    });

    await act(async () => {
      await fireEvent.changeText(username, '');
      await fireEvent.changeText(password, '');
    });

    expect(loginButton.props.accessibilityState).toStrictEqual({
      disabled: true,
    });
  });

//   test('Button is disabled when API is called', async () => {
//     const {getByTestId} = render(
//       <Provider store={store}>
//         <PaperProvider>
//           <LoginScreen />
//         </PaperProvider>
//       </Provider>,
//     );

//     const username = getByTestId('username');
//     const password = getByTestId('password');
//     const loginButton = getByTestId('loginButton');

//     await act(async () => {
//       await fireEvent.changeText(username, 'test');
//       await fireEvent.changeText(password, 'test');
//     });

//     console.log(loginButton.props);

//     expect(loginButton.props.accessibilityState).toStrictEqual({
//       disabled: false,
//     });

//     await act(async () => {
//       await fireEvent.press(loginButton);
//     });

//     await waitFor(() => {
//       console.log(loginButton.props.accessibilityState);
//       expect(loginButton.props.accessibilityState).toStrictEqual({
//         disabled: true,
//       });
//     });
//   });
});
