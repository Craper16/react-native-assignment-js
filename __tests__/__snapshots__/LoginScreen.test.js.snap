// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginScreen Screen renders correctly 1`] = `
<RNCSafeAreaProvider
  onInsetsChange={[Function]}
  style={
    [
      {
        "flex": 1,
      },
      undefined,
    ]
  }
>
  <View
    collapsable={false}
    pointerEvents="box-none"
    style={
      {
        "flex": 1,
      }
    }
  >
    <View
      onLayout={[Function]}
      style={
        [
          {
            "backgroundColor": "#aab1eebb",
            "flex": 1,
          },
          {
            "paddingBottom": 0,
          },
        ]
      }
    >
      <RNGestureHandlerButton
        collapsable={false}
        delayLongPress={600}
        enabled={true}
        exclusive={true}
        handlerTag={1}
        handlerType="NativeViewGestureHandler"
        onGestureEvent={[Function]}
        onGestureHandlerEvent={[Function]}
        onGestureHandlerStateChange={[Function]}
        onHandlerStateChange={[Function]}
        rippleColor={0}
        touchSoundDisabled={false}
      >
        <View
          accessible={true}
          collapsable={false}
          style={
            {
              "alignItems": "center",
              "justifyContent": "center",
              "marginVertical": 100,
            }
          }
        >
          <Text
            style={
              [
                {
                  "textAlign": "left",
                },
                {
                  "color": "rgba(28, 27, 31, 1)",
                  "fontFamily": "System",
                  "fontWeight": "400",
                  "letterSpacing": 0,
                },
                {
                  "writingDirection": "ltr",
                },
                {
                  "color": "#8A2BE2",
                  "fontSize": 24,
                  "fontWeight": "bold",
                  "marginBottom": 16,
                },
              ]
            }
          >
            Login to view Articles!
          </Text>
          <View
            style={
              {
                "marginBottom": 9,
                "marginVertical": 14,
                "padding": 2,
                "width": "80%",
              }
            }
          >
            <RCTScrollView>
              <View>
                <View
                  style={
                    {
                      "justifyContent": "center",
                      "margin": 5,
                      "padding": 4,
                    }
                  }
                >
                  <View
                    style={{}}
                  >
                    <View
                      pointerEvents="none"
                      style={
                        [
                          {
                            "bottom": 0,
                            "left": 0,
                            "position": "absolute",
                            "right": 0,
                            "top": 6,
                          },
                          {
                            "backgroundColor": "rgba(255, 251, 254, 1)",
                            "borderColor": "rgba(121, 116, 126, 1)",
                            "borderRadius": 4,
                            "borderWidth": 1,
                          },
                          undefined,
                        ]
                      }
                      testID="text-input-outline"
                    />
                    <View>
                      <View
                        style={
                          [
                            {
                              "paddingBottom": 0,
                            },
                            {
                              "minHeight": 56,
                              "paddingTop": 8,
                            },
                          ]
                        }
                      >
                        <View
                          collapsable={false}
                          pointerEvents="none"
                          style={
                            {
                              "bottom": 0,
                              "left": 0,
                              "opacity": 1,
                              "position": "absolute",
                              "right": 0,
                              "top": 0,
                              "transform": [
                                {
                                  "translateX": 0,
                                },
                              ],
                              "zIndex": 3,
                            }
                          }
                        >
                          <View
                            collapsable={false}
                            pointerEvents="none"
                            style={
                              {
                                "backgroundColor": "rgba(255, 251, 254, 1)",
                                "bottom": 4,
                                "left": 10,
                                "maxHeight": 2,
                                "opacity": 0,
                                "position": "absolute",
                                "right": 0,
                                "top": 6,
                                "transform": [
                                  {
                                    "translateX": 0,
                                  },
                                ],
                                "width": 12,
                              }
                            }
                          />
                          <Text
                            collapsable={false}
                            maxFontSizeMultiplier={1.5}
                            numberOfLines={1}
                            style={
                              {
                                "backgroundColor": "rgba(255, 251, 254, 1)",
                                "color": "transparent",
                                "fontFamily": "System",
                                "fontSize": 16,
                                "fontWeight": undefined,
                                "left": 8,
                                "letterSpacing": 0.15,
                                "lineHeight": undefined,
                                "opacity": 0,
                                "paddingHorizontal": 0,
                                "position": "absolute",
                                "textAlign": "left",
                                "top": 33,
                                "transform": [
                                  {
                                    "translateX": 0,
                                  },
                                  {
                                    "translateY": 0,
                                  },
                                  {
                                    "scale": 1,
                                  },
                                  {
                                    "scaleY": 1,
                                  },
                                ],
                                "width": -16,
                                "writingDirection": "ltr",
                              }
                            }
                          >
                            Username
                          </Text>
                          <Text
                            collapsable={false}
                            maxFontSizeMultiplier={1.5}
                            numberOfLines={1}
                            onLayout={[Function]}
                            style={
                              {
                                "color": "rgba(103, 80, 164, 1)",
                                "fontFamily": "System",
                                "fontSize": 16,
                                "fontWeight": undefined,
                                "left": 0,
                                "letterSpacing": 0.15,
                                "lineHeight": undefined,
                                "opacity": 0,
                                "paddingHorizontal": 16,
                                "position": "absolute",
                                "textAlign": "left",
                                "top": 32,
                                "transform": [
                                  {
                                    "translateX": 0,
                                  },
                                  {
                                    "translateY": 0,
                                  },
                                  {
                                    "scale": 1,
                                  },
                                ],
                                "writingDirection": "ltr",
                              }
                            }
                            testID="username-label-active"
                          >
                            Username
                          </Text>
                          <Text
                            collapsable={false}
                            maxFontSizeMultiplier={1.5}
                            numberOfLines={1}
                            style={
                              {
                                "color": "rgba(73, 69, 79, 1)",
                                "fontFamily": "System",
                                "fontSize": 16,
                                "fontWeight": undefined,
                                "left": 0,
                                "letterSpacing": 0.15,
                                "lineHeight": undefined,
                                "opacity": 0,
                                "paddingHorizontal": 16,
                                "position": "absolute",
                                "textAlign": "left",
                                "top": 32,
                                "transform": [
                                  {
                                    "translateX": 0,
                                  },
                                  {
                                    "translateY": 0,
                                  },
                                  {
                                    "scale": 1,
                                  },
                                ],
                                "writingDirection": "ltr",
                              }
                            }
                            testID="username-label-inactive"
                          >
                            Username
                          </Text>
                        </View>
                        <TextInput
                          autoCapitalize={false}
                          editable={true}
                          maxFontSizeMultiplier={1.5}
                          multiline={false}
                          onBlur={[Function]}
                          onChangeText={[Function]}
                          onFocus={[Function]}
                          placeholder=" "
                          placeholderTextColor="rgba(73, 69, 79, 1)"
                          selectionColor="rgba(103, 80, 164, 1)"
                          style={
                            [
                              {
                                "margin": 0,
                                "zIndex": 1,
                              },
                              {
                                "height": 48,
                              },
                              {
                                "paddingBottom": 0,
                                "paddingTop": 0,
                              },
                              {
                                "color": "#8A2BE2",
                                "fontFamily": "System",
                                "fontSize": 16,
                                "fontWeight": undefined,
                                "letterSpacing": 0.15,
                                "lineHeight": undefined,
                                "paddingHorizontal": 16,
                                "textAlign": "left",
                                "textAlignVertical": "center",
                              },
                              false,
                              [
                                {},
                              ],
                              undefined,
                            ]
                          }
                          testID="username"
                          underlineColorAndroid="transparent"
                          value=""
                        />
                      </View>
                    </View>
                  </View>
                </View>
                <View
                  style={
                    {
                      "justifyContent": "center",
                      "margin": 5,
                      "padding": 4,
                    }
                  }
                >
                  <View
                    style={{}}
                  >
                    <View
                      pointerEvents="none"
                      style={
                        [
                          {
                            "bottom": 0,
                            "left": 0,
                            "position": "absolute",
                            "right": 0,
                            "top": 6,
                          },
                          {
                            "backgroundColor": "rgba(255, 251, 254, 1)",
                            "borderColor": "rgba(121, 116, 126, 1)",
                            "borderRadius": 4,
                            "borderWidth": 1,
                          },
                          undefined,
                        ]
                      }
                      testID="text-input-outline"
                    />
                    <View>
                      <View
                        style={
                          [
                            {
                              "paddingBottom": 0,
                            },
                            {
                              "minHeight": 56,
                              "paddingTop": 8,
                            },
                          ]
                        }
                      >
                        <View
                          collapsable={false}
                          pointerEvents="none"
                          style={
                            {
                              "bottom": 0,
                              "left": 0,
                              "opacity": 1,
                              "position": "absolute",
                              "right": 0,
                              "top": 0,
                              "transform": [
                                {
                                  "translateX": 0,
                                },
                              ],
                              "zIndex": 3,
                            }
                          }
                        >
                          <View
                            collapsable={false}
                            pointerEvents="none"
                            style={
                              {
                                "backgroundColor": "rgba(255, 251, 254, 1)",
                                "bottom": 4,
                                "left": 10,
                                "maxHeight": 2,
                                "opacity": 0,
                                "position": "absolute",
                                "right": 0,
                                "top": 6,
                                "transform": [
                                  {
                                    "translateX": 0,
                                  },
                                ],
                                "width": 12,
                              }
                            }
                          />
                          <Text
                            collapsable={false}
                            maxFontSizeMultiplier={1.5}
                            numberOfLines={1}
                            style={
                              {
                                "backgroundColor": "rgba(255, 251, 254, 1)",
                                "color": "transparent",
                                "fontFamily": "System",
                                "fontSize": 16,
                                "fontWeight": undefined,
                                "left": 8,
                                "letterSpacing": 0.15,
                                "lineHeight": undefined,
                                "opacity": 0,
                                "paddingHorizontal": 0,
                                "position": "absolute",
                                "textAlign": "left",
                                "top": 33,
                                "transform": [
                                  {
                                    "translateX": 0,
                                  },
                                  {
                                    "translateY": 0,
                                  },
                                  {
                                    "scale": 1,
                                  },
                                  {
                                    "scaleY": 1,
                                  },
                                ],
                                "width": -16,
                                "writingDirection": "ltr",
                              }
                            }
                          >
                            Password
                          </Text>
                          <Text
                            collapsable={false}
                            maxFontSizeMultiplier={1.5}
                            numberOfLines={1}
                            onLayout={[Function]}
                            style={
                              {
                                "color": "rgba(103, 80, 164, 1)",
                                "fontFamily": "System",
                                "fontSize": 16,
                                "fontWeight": undefined,
                                "left": 0,
                                "letterSpacing": 0.15,
                                "lineHeight": undefined,
                                "opacity": 0,
                                "paddingHorizontal": 16,
                                "position": "absolute",
                                "textAlign": "left",
                                "top": 32,
                                "transform": [
                                  {
                                    "translateX": 0,
                                  },
                                  {
                                    "translateY": 0,
                                  },
                                  {
                                    "scale": 1,
                                  },
                                ],
                                "writingDirection": "ltr",
                              }
                            }
                            testID="password-label-active"
                          >
                            Password
                          </Text>
                          <Text
                            collapsable={false}
                            maxFontSizeMultiplier={1.5}
                            numberOfLines={1}
                            style={
                              {
                                "color": "rgba(73, 69, 79, 1)",
                                "fontFamily": "System",
                                "fontSize": 16,
                                "fontWeight": undefined,
                                "left": 0,
                                "letterSpacing": 0.15,
                                "lineHeight": undefined,
                                "opacity": 0,
                                "paddingHorizontal": 16,
                                "position": "absolute",
                                "textAlign": "left",
                                "top": 32,
                                "transform": [
                                  {
                                    "translateX": 0,
                                  },
                                  {
                                    "translateY": 0,
                                  },
                                  {
                                    "scale": 1,
                                  },
                                ],
                                "writingDirection": "ltr",
                              }
                            }
                            testID="password-label-inactive"
                          >
                            Password
                          </Text>
                        </View>
                        <TextInput
                          autoCapitalize={false}
                          editable={true}
                          maxFontSizeMultiplier={1.5}
                          multiline={false}
                          onBlur={[Function]}
                          onChangeText={[Function]}
                          onFocus={[Function]}
                          placeholder=" "
                          placeholderTextColor="rgba(73, 69, 79, 1)"
                          secureTextEntry={true}
                          selectionColor="rgba(103, 80, 164, 1)"
                          style={
                            [
                              {
                                "margin": 0,
                                "zIndex": 1,
                              },
                              {
                                "height": 48,
                              },
                              {
                                "paddingBottom": 0,
                                "paddingTop": 0,
                              },
                              {
                                "color": "#8A2BE2",
                                "fontFamily": "System",
                                "fontSize": 16,
                                "fontWeight": undefined,
                                "letterSpacing": 0.15,
                                "lineHeight": undefined,
                                "paddingHorizontal": 16,
                                "textAlign": "left",
                                "textAlignVertical": "center",
                              },
                              false,
                              [
                                {},
                              ],
                              undefined,
                            ]
                          }
                          testID="password"
                          underlineColorAndroid="transparent"
                          value=""
                        />
                      </View>
                    </View>
                  </View>
                </View>
                <View
                  style={
                    {
                      "alignItems": "center",
                      "height": 48,
                      "justifyContent": "center",
                      "width": "100%",
                    }
                  }
                >
                  <View
                    collapsable={false}
                    style={
                      {
                        "alignSelf": undefined,
                        "bottom": undefined,
                        "left": undefined,
                        "position": undefined,
                        "right": undefined,
                        "shadowColor": "#000",
                        "shadowOffset": {
                          "height": 1,
                          "width": 0,
                        },
                        "shadowOpacity": 0.15,
                        "shadowRadius": 3,
                        "top": undefined,
                      }
                    }
                  >
                    <View
                      collapsable={false}
                      style={
                        {
                          "shadowColor": "#000",
                          "shadowOffset": {
                            "height": 1,
                            "width": 0,
                          },
                          "shadowOpacity": 0.3,
                          "shadowRadius": 1,
                        }
                      }
                    >
                      <View
                        collapsable={false}
                        style={
                          {
                            "backgroundColor": "rgb(247, 243, 249)",
                            "borderColor": "transparent",
                            "borderRadius": 20,
                            "borderStyle": "solid",
                            "borderWidth": 0,
                            "minWidth": 64,
                          }
                        }
                      >
                        <View
                          accessibilityRole="button"
                          accessibilityState={
                            {
                              "disabled": false,
                            }
                          }
                          accessible={true}
                          collapsable={false}
                          focusable={true}
                          onBlur={[Function]}
                          onClick={[Function]}
                          onFocus={[Function]}
                          onResponderGrant={[Function]}
                          onResponderMove={[Function]}
                          onResponderRelease={[Function]}
                          onResponderTerminate={[Function]}
                          onResponderTerminationRequest={[Function]}
                          onStartShouldSetResponder={[Function]}
                          style={
                            [
                              {
                                "overflow": "hidden",
                              },
                              false,
                              {
                                "borderRadius": 20,
                              },
                            ]
                          }
                          testID="loginButton"
                        >
                          <View
                            style={
                              [
                                {
                                  "alignItems": "center",
                                  "flexDirection": "row",
                                  "justifyContent": "center",
                                },
                                undefined,
                              ]
                            }
                          >
                            <Text
                              numberOfLines={1}
                              selectable={false}
                              style={
                                [
                                  {
                                    "color": "rgba(28, 27, 31, 1)",
                                    "fontFamily": "System",
                                    "fontSize": 14,
                                    "fontWeight": "500",
                                    "letterSpacing": 0.1,
                                    "lineHeight": 20,
                                  },
                                  {
                                    "textAlign": "left",
                                  },
                                  {
                                    "writingDirection": "ltr",
                                  },
                                  [
                                    {
                                      "marginHorizontal": 16,
                                      "marginVertical": 9,
                                      "textAlign": "center",
                                    },
                                    false,
                                    {
                                      "marginHorizontal": 24,
                                      "marginVertical": 10,
                                    },
                                    undefined,
                                    false,
                                    {
                                      "color": "rgba(103, 80, 164, 1)",
                                      "fontFamily": "System",
                                      "fontSize": 14,
                                      "fontWeight": "500",
                                      "letterSpacing": 0.1,
                                      "lineHeight": 20,
                                    },
                                    undefined,
                                  ],
                                ]
                              }
                            >
                              Login
                            </Text>
                          </View>
                        </View>
                      </View>
                    </View>
                  </View>
                </View>
              </View>
            </RCTScrollView>
          </View>
        </View>
      </RNGestureHandlerButton>
    </View>
  </View>
</RNCSafeAreaProvider>
`;
